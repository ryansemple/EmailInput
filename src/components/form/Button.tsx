import React from "react";

interface ButtonProps {
	text: string,
	onClick: Function,
	style?: any,
	disabled?: boolean
}

const Button = (props: ButtonProps) => {

	const onClick = 
	(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => 
	{
		props.onClick();
	};

	return (
		<button
			type="button"
			className="Button"
			onClick={onClick}
			style={props.style}
			disabled={props.disabled}>
			{props.text}
		</button>
	);
}

export default Button;