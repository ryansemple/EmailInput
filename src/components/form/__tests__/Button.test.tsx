import React from "react";
import { render } from "@testing-library/react";
import Button from "../Button";

test("Renders Button component correctly.", () => {
	const { container } = render
	(
		<Button 
			text="Text"
			onClick={() => alert("clicked") }
			style={{
				color: "red"
			}}
			disabled={true}
			disabledTooltipTitle="disabled"
			className="class"
		/>
	);
	expect(container).toMatchSnapshot();
});