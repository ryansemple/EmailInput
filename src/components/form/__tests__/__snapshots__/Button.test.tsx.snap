// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button component correctly. 1`] = `
<div>
  <button
    class="class Button"
    disabled=""
    style="color: red;"
    title="disabled"
    type="button"
  >
    Text
  </button>
</div>
`;
